decidim-core/app/services/decidim/resource_search.rb

```diff
@@ -39,12 +39,21 @@ def search_category_id
     # `"global"` as parameter, and in the method we do the needed changes to search
     # properly.
     #
-    # You can use the `search_organization_scopes` helper method, defined in
-    # `Decidim::OrganizationScopesHelper`, to render the collection needed for the
+    # You can use the `search_participatory_process_subscopes` helper method, defined in
+    # `Decidim::ParticipatoryProcessScopesHelper`, to render the collection needed for the
     # `collection_check_boxes` form method.
     def search_scope_id
-      clean_scope_ids = [scope_id].flatten.map { |id| id == "global" ? nil : id }
-      query.where(decidim_scope_id: clean_scope_ids)
+      clean_scope_ids = [scope_id].flatten
+
+      conditions = []
+      conditions << "decidim_scope_id IS NULL" if clean_scope_ids.delete("global")
```

Comment:
I think the right check here would be to check the other table: `decidim_scopes: { id: nil }`. Otherwise, if there are any orphan records, they'll be included in the result. If there was a foreign key between from `decidim_scope_id` pointing to the `scopes` table it would be equivalent I think. But the standard way to do it is to check the joined table anyways.